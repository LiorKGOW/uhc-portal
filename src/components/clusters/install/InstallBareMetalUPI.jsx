import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  PageSection,
} from '@patternfly/react-core';

import Breadcrumbs from '../../common/Breadcrumbs';
import { tollboothActions } from '../../../redux/actions';
import { scrollToTop } from '../../../common/helpers';
import instructionsMapping from './instructions/instructionsMapping';
import OCPInstructions from './instructions/OCPInstructions';
import PageTitle from '../../common/PageTitle';

export class InstallBareMetalUPI extends Component {
  componentDidMount() {
    scrollToTop();
    document.title = 'Install OpenShift 4 | Red Hat OpenShift Cluster Manager | Bare Metal User-Provisioned Infrastructure';

    const { dispatch } = this.props;
    dispatch(tollboothActions.createAuthToken());
  }

  render() {
    const { token } = this.props;
    const breadcrumbs = (
      <Breadcrumbs path={[
        { label: 'Clusters' },
        { label: 'Create', path: '/create' },
        { label: 'Bare Metal', path: '/install/metal' },
        { label: 'User-provisioned infrastructure' },
      ]}
      />
    );

    return (
      <>
        <PageTitle title={instructionsMapping.baremetal.x86.upi.title} breadcrumbs={breadcrumbs} />
        <PageSection>
          <OCPInstructions
            token={token}
            breadcrumbs={breadcrumbs}
            cloudProviderID="baremetal"
            customizations={instructionsMapping.baremetal.customizations}
            {...instructionsMapping.baremetal.x86.upi}
          />
        </PageSection>
      </>
    );
  }
}

InstallBareMetalUPI.propTypes = {
  token: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({ token: state.tollbooth.token });

export default connect(mapStateToProps)(InstallBareMetalUPI);
