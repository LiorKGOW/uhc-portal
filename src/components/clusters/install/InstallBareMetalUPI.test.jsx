import React from 'react';
import { shallow } from 'enzyme';

import { InstallBareMetalUPI } from './InstallBareMetalUPI';

describe('BareMetal UPI install', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<InstallBareMetalUPI dispatch={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
