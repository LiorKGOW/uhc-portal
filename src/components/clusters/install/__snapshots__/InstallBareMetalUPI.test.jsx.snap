// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BareMetal UPI install renders correctly 1`] = `
<Fragment>
  <PageTitle
    breadcrumbs={
      <Breadcrumbs
        path={
          Array [
            Object {
              "label": "Clusters",
            },
            Object {
              "label": "Create",
              "path": "/create",
            },
            Object {
              "label": "Bare Metal",
              "path": "/install/metal",
            },
            Object {
              "label": "User-provisioned infrastructure",
            },
          ]
        }
      />
    }
    title="Install OpenShift on Bare Metal with user-provisioned infrastructure"
  />
  <PageSection>
    <OCPInstructions
      architecture="x86"
      breadcrumbs={
        <Breadcrumbs
          path={
            Array [
              Object {
                "label": "Clusters",
              },
              Object {
                "label": "Create",
                "path": "/create",
              },
              Object {
                "label": "Bare Metal",
                "path": "/install/metal",
              },
              Object {
                "label": "User-provisioned infrastructure",
              },
            ]
          }
        />
      }
      channel="stable"
      cloudProviderID="bareMetal"
      customizations="https://docs.openshift.com/container-platform/latest/installing/installing_bare_metal/installing-bare-metal-network-customizations.html"
      displayRHCOSSection={true}
      docURL="https://docs.openshift.com/container-platform/latest/installing/installing_bare_metal/installing-bare-metal.html"
      isBMIPI={false}
      rhcosAdditionalInstructions="Download the installer ISO image and the compressed metal RAW."
      rhcosDownloads={
        Array [
          Array [
            Object {
              "archURL": Object {
                "ppc": "https://mirror.openshift.com/pub/openshift-v4/ppc64le/dependencies/rhcos/latest/latest/rhcos-live.ppc64le.iso",
                "s390x": "https://mirror.openshift.com/pub/openshift-v4/s390x/dependencies/rhcos/latest/latest/rhcos-live.s390x.iso",
                "x86": "https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/latest/latest/rhcos-live.x86_64.iso",
              },
              "buttonText": "Download RHCOS ISO",
              "name": "OCP-Download-RHCOS-ISO",
            },
            Object {
              "archURL": Object {
                "ppc": "https://mirror.openshift.com/pub/openshift-v4/ppc64le/dependencies/rhcos/latest/latest/rhcos-metal.ppc64le.raw.gz",
                "s390x": "https://mirror.openshift.com/pub/openshift-v4/s390x/dependencies/rhcos/latest/latest/rhcos-metal.s390x.raw.gz",
                "x86": "https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/latest/latest/rhcos-metal.x86_64.raw.gz",
              },
              "buttonText": "Download RHCOS RAW",
              "name": "OCP-Download-RHCOS-RAW",
            },
          ],
        ]
      }
      rhcosLearnMoreURL="https://docs.openshift.com/container-platform/latest/installing/installing_bare_metal/installing-bare-metal.html#creating-machines-bare-metal"
      title="Install OpenShift on Bare Metal with user-provisioned infrastructure"
      token={Object {}}
    />
  </PageSection>
</Fragment>
`;
